app/models/storage_location_repository_row.rb

```diff
@@ -19,6 +19,8 @@ class StorageLocationRepositoryRow < ApplicationRecord
     validate :unique_repository_row
   end
 
+  validate :must_belong_to_container
```

Comment:
if it is always required - just define a scoped belongs_to relation:

belongs_to :container_storage_location,
             -> { where(container: true) },
             class_name: 'StorageLocation',
             foreign_key: 'storage_location_id',
             inverse_of: :storage_location_repository_rows